[Feat] #50 장소정보를 보내는 기능을 작성했어요!

suhyeon7497 opened the pull request:
### ✨ 작업 내용
 - place-message/{place-id} url로 get 요청을 보내면 나에게 보내기로 장소 정보를 받아올 수 있습니다.
---

### ✨ 참고 사항
없음.
---

### ⏰ 현재 버그

---

### ✏ Git Close

suhyeon7497 committed:
[feat] placeMessage를 위한 LinkTemplate정의

LinkTemplate를 통해 kakao api와 통신.

관련 이슈: #50

suhyeon7497 committed:
[feat] placeMessage를 보내는 기능 작성

placeMessage를 보내는 기능을 작성하였습니다.

관련 이슈: #50

suhyeon7497 committed:
[feat] placeMessage를 보내는 controller를 작성하였습니다.

placeMessage를 보내는 Controller를 작성하였습니다.

관련 이슈: #50

sanghee0820 commented on src/main/java/team7/inplace/placeMessage/application/command/PlaceMessageCommand.java:
```diff
@@ -0,0 +1,44 @@
+        );
+    }
+
+    public static String videoUrlToImgUrl(String videoUrl) {
+        Matcher matcher = Pattern.compile(REGEX).matcher(videoUrl);
+
+        if (matcher.find()) {
+            String videoId = matcher.group(1);
+            return String.format("https://img.youtube.com/vi/%s/maxresdefault.jpg", videoId);
+        } else {
+            return null;
+        }
+    }
```
이 부분을, 굳이 REGEX로 파싱하지 않아도 될 것 같습니다.
video에서 getUUID 메서드를 이용하면 UUID를 바로 가져올 수 잇어요!

suhyeon7497 replied:
이미 동윤님이 만드신 PlaceDetailInfo를 재사용해보려했는데, 오히려 uuid -> url -> uuid로 가는 좋지 않은 로직이 되어버린 것 같습니다. 수정하도록 하겠습니다.

sanghee0820 commented on src/main/java/team7/inplace/placeMessage/util/KakaoMessageMaker.java:
```diff
@@ -2,27 +2,29 @@
 @RequiredArgsConstructor
 public class KakaoMessageMaker {
 
+    public static final String TEMPLATE_OBJECT = "template_object";
```
이 ㅂ변수를 다른데서 사용하는 경우가 있나요?

suhyeon7497 replied:
template_object는 모든 카카오 메세지 형식에서 사용됩니다. 향후에 다른 메세지 타입이 생길 수 있으니 카카오 메세지를 만들어주는 kakaoMessageMaker에 넣어두면 좋을 것이라고 생각하였습니다.

sanghee0820 reviewed (changes requested):
어려운 기능이었을텐데 고생하셨습니다!
match하는 부분만 조금 바꿔주시면 좋을 것 같아요!

suhyeon7497 committed:
[refactor] placeMessageCommand에서 uuid를 바로 사용하도록 하였습니다.

기존에 작성되어있는 것을 사용했을 때 uuid -> videoUrl -> uuid 내부 로직을 uuid 그대로 사용하도록 하욨습니다.

관련 이슈: #50

suhyeon7497 committed:
Merge branch 'weekly/10' of https://github.com/kakao-tech-campus-2nd-step3/Team7_BE into feat/#50-placeMessage

dong-yxxn reviewed (approved):
수고많으셨습니다

suhyeon7497 committed:
[feat] 특정 시간 뒤 작동하는 기능

특정 시간 이후에 작동하는 기능을 작성하였습니다.

관련 이슈: #50

dong-yxxn committed:
Merge branch 'weekly/10' into feat/#50-placeMessage

suhyeon7497 committed:
[feat] 3일 뒤 review message를 보내는 기능

3일 뒤에 review message를 보내는 기능을 작성하였습니다.

관련 이슈: #50

suhyeon7497 committed:
Merge branch 'weekly/10' of https://github.com/kakao-tech-campus-2nd-step3/Team7_BE into feat/#50-placeMessage

suhyeon7497 committed:
Merge branch 'feat/#50-placeMessage' of https://github.com/kakao-tech-campus-2nd-step3/Team7_BE into feat/#50-placeMessage
